### PR TITLE
Beds: Improve force-night-skip input checks

### PR DESCRIPTION
**Pull request by paramat**
Attends to #1964 
WIP seems to have a bug, can't force night skip on a local server with 3 players.

### CODE REVIEW

**Comment by paramat** [comment]
Ok with 2 players in bed, formspec shows '2 of 3 players in bed' but printing out 'player_in_bed' shows 1.

File: mods/beds/functions.lua [hunk 1]
```diff
@@ -214,8 +214,9 @@ minetest.register_on_player_receive_fields(function(player, formname, fields)
 	end
 
 	if fields.force then
-		update_formspecs(is_night_skip_enabled())
-		if is_night_skip_enabled() then
+		local is_majority = (#minetest.get_connected_players() / 2) < player_in_bed
```
Comment (SmallJoker) [inline mods/beds/functions.lua]: Should be in a new function to have this calculation the same wherever it is required.

Comment (paramat) [reply mods/beds/functions.lua]: Yes. Will do if we manage to fix the problem.

**Comment by paramat** [comment]
Testing current master with 3 players on a local server force night skip works, so the problem is in my PR somehow. maybe this is why the code is how it is? Anyway i'm stuck with this now.

**Comment by paramat** [comment]
Stuck.

**Comment by paramat** [comment]
Best someone else works on this, i can't be bothered, @sofar is the formspec abuse and beds mod expert.

**Comment by bell07** [comment]
Maybe the issue is connected to #2111? Did you looked to the counter in your tests?

**Pull request closed without merging**